lib/modules/reducer.js

```diff
@@ -22,7 +22,8 @@ export default (state=DEFAULT, action={}) => {
     }
 
     case actions.SET_PAGE: {
-      const referrerFromHistory = !isEmpty(state.currentPage)
+      const historyLen = state.history.length;
+      const referrerFromHistory = !isEmpty(state.currentPage) && historyLen > 1
```

Comment:
I don't want to embed this magic idea of "2 or more items in history" in our codebase when that idea stems from a bug.